examples/linear_model/plot_quantile_regression.py

```diff
@@ -11,100 +12,306 @@
+# target `y` while keeping the same expected value:
+#
+# - in the first case, a heteroscedastic Normal noise is added;
+# - in the second case, an asymmetric Pareto noise is added.
+y_normal = y_true_mean + rng.normal(
+    loc=0, scale=0.5 + 0.5 * x, size=x.shape[0]
+)
+a = 5
+y_pareto = y_true_mean + 10 * (rng.pareto(a, size=x.shape[0]) - 1 / (a - 1))
+
+# %%
+# Let's first visualize the datasets as well as the distribution of the
+# residuals `y - mean(y)`.
 import matplotlib.pyplot as plt
 
-from sklearn.linear_model import QuantileRegressor, LinearRegression
-from sklearn.metrics import mean_absolute_error, mean_squared_error
-from sklearn.model_selection import cross_val_score
+_, axs = plt.subplots(
+    nrows=2, ncols=2, figsize=(15, 11), sharex="row", sharey="row"
+)
 
+axs[0, 0].plot(x, y_true_mean, label="True mean")
+axs[0, 0].scatter(
+    x, y_normal, color="black", alpha=0.5, label="Observations"
+)
+axs[1, 0].hist(y_true_mean - y_normal, edgecolor="black")
 
-def plot_points_highlighted(x, y, model_low, model_high, ax):
-    """Plot points with highlighting."""
-    mask = y <= model_low.predict(X)
-    ax.scatter(x[mask], y[mask], c="k", marker="x")
-    mask = y > model_high.predict(X)
-    ax.scatter(x[mask], y[mask], c="k", marker="x")
-    mask = (y > model_low.predict(X)) & (y <= model_high.predict(X))
-    ax.scatter(x[mask], y[mask], c="k")
 
+axs[0, 1].plot(x, y_true_mean, label="True mean")
+axs[0, 1].scatter(
+    x, y_pareto, color="black", alpha=0.5, label="Observations"
+)
+axs[1, 1].hist(y_true_mean - y_pareto, edgecolor="black")
 
-fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(15, 5), sharey=True)
+axs[0, 0].set_title("Dataset with heteroscedastic Normal distributed targets")
+axs[0, 1].set_title("Dataset with asymmetric Pareto distributed target")
+axs[1, 0].set_title(
+    "Residuals distribution for heteroscedastic Normal distributed targets"
+)
+axs[1, 1].set_title(
+    "Residuals distribution for asymmetric Pareto distributed target"
+)
+axs[0, 0].legend()
+axs[0, 1].legend()
+axs[0, 0].set_ylabel("y")
+axs[1, 0].set_ylabel("Counts")
+axs[0, 1].set_xlabel("x")
+axs[0, 0].set_xlabel("x")
+axs[1, 0].set_xlabel("Residuals")
+_ = axs[1, 1].set_xlabel("Residuals")
 
-rng = np.random.RandomState(42)
-x = np.linspace(0, 10, 100)
-X = x[:, np.newaxis]
-y = 10 + 0.5 * x + rng.normal(loc=0, scale=0.5 + 0.5 * x, size=x.shape[0])
-y_mean = 10 + 0.5 * x
-ax1.plot(x, y_mean, "k--")
+# %%
+# With the heteroscedastic Normal distributed target, we observe that the
+# variance of the noise is increasing when the value of the feature `x` is
+# increasing.
+#
+# With the asymmetric Pareto distributed target, we observe that the positive
+# residuals are bounded.
+#
+# These types of noisy targets violate one of the key assumption made when one
+# wants to use a :class:`~sklearn.linear_model.LinearRegression`: the noise is
+# expected to be normally distributed with a constant variance.
```

Comment:
This needs adaption. Normally and homoscedastic distributed targets assert that `coef ~ Normal(...)`. We don't do inference on the coefficients (I don't even dare linking to the corresponding issue:smirk:). Maybe
```suggestion
# These types of noisy targets make the estimation via :class:`~sklearn.linear_model.LinearRegression`
# less efficient, i.e. we need more data to get stable results and, in addition, large outliers can have a huge
# impact on the fitted coeffients.
# (Stated otherwise: in a setting with constant variance, ordinary least squares estimators converge much faster
# to the *true* coefficients with increasing sample size.)
```